crates/aiken-lang/src/gen_uplc.rs

```diff
@@ -4083,61 +4083,17 @@ impl<'a> CodeGenerator<'a> {
                 ValueConstructorVariant::ModuleConstant { .. } => {
                     unreachable!("{:#?}, {}", constructor, name)
                 }
-
-                ValueConstructorVariant::ModuleFn {
-                    builtin: Some(builtin),
-                    ..
-                } => {
-                    let term = match builtin {
-                        DefaultFunction::IfThenElse
-                        | DefaultFunction::ChooseUnit
-                        | DefaultFunction::Trace
-                        | DefaultFunction::ChooseList
-                        | DefaultFunction::ChooseData
-                        | DefaultFunction::UnConstrData => {
-                            builder::special_case_builtin(builtin, 0, vec![])
-                        }
-
-                        DefaultFunction::FstPair | DefaultFunction::SndPair => {
-                            builder::undata_builtin(
-                                builtin,
-                                0,
-                                &constructor.tipo.return_type().unwrap(),
-                                vec![],
-                            )
-                        }
-
-                        DefaultFunction::HeadList
-                            if !constructor.tipo.return_type().unwrap().is_pair() =>
-                        {
-                            builder::undata_builtin(
-                                builtin,
-                                0,
-                                &constructor.tipo.return_type().unwrap(),
-                                vec![],
-                            )
-                        }
-
-                        DefaultFunction::MkCons | DefaultFunction::MkPairData => {
-                            unimplemented!(
-                                "MkCons and MkPairData should be handled by an anon function or using [] or ( a, b, .., z) or Pair {{fst:a, snd: b}}.\n"
-                            )
-                        }
-                        _ => {
-                            let mut term: Term<Name> = (*builtin).into();
-
-                            term = builder::apply_builtin_forces(term, builtin.force_count());
-
-                            term
-                        }
-                    };
-                    Some(term)
-                }
```

Comment:
Note: this is actually never reached because any builtin call should have been replaced by an appropriate Air::Builtin expression by then.  Added an assertion in the next branch to capture that assumption.

Reply:
This can be reached if you assign a builtin to a variable and then use that variable

Reply:
Indeed, I noticed as one of the acceptance test failed. It only happens if you pass the builtin as a variable to a function, but not if you call it in place. 

Anyway, the comment is outdated, as I've replaced the assertion with a fallback.